WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaPickerHelper.swift

```diff
@@ -109,17 +110,47 @@ class GutenbergMediaPickerHelper: NSObject {
     }
 }
 
+// MARK: - User messages for video limits allowances
+//
+extension GutenbergMediaPickerHelper: VideoLimitsAlertPresenter {}
+
+// MARK: - Picker Delegate
+//
 extension GutenbergMediaPickerHelper: WPMediaPickerViewControllerDelegate {
 
     func mediaPickerController(_ picker: WPMediaPickerViewController, didFinishPicking assets: [WPMediaAsset]) {
-        invokeMediaPickerCallback(asset: assets)
-        picker.dismiss(animated: true, completion: nil)
+        if picker == self.cameraPicker,
+           let asset = assets.first,
+           asset.exceedsFreeSitesAllowance(),
+           !self.post.blog.hasPaidPlan {
+                self.presentVideoLimitExceededAfterCapture(on: self.context)
+        } else {
+            invokeMediaPickerCallback(asset: assets)
+            picker.dismiss(animated: true, completion: nil)
+        }
     }
 
     func mediaPickerControllerDidCancel(_ picker: WPMediaPickerViewController) {
         context.dismiss(animated: true, completion: { self.invokeMediaPickerCallback(asset: nil) })
     }
 
+    func mediaPickerController(_ picker: WPMediaPickerViewController, shouldShowOverlayViewForCellFor asset: WPMediaAsset) -> Bool {
+        picker !== cameraPicker &&
+        asset.exceedsFreeSitesAllowance() &&
+        !post.blog.hasPaidPlan
```

Comment:
We're repeating this bit of logic in three places in this class – perhaps we should create a smaller helper method?

```
func canUploadAsset(_ asset: WPMediaAsset, for blog: Blog) -> Bool {
    asset.exceeedsFreeSitesAllowance() && !blog.hasPaidPlan
}
```

Reply:
Or thinking about it, we're using this same bit of logic across a whole range of classes multiple times. Perhaps we should move it onto `Blog`? `canUploadAsset(_ asset: WPMediaAsset)`?

Reply:
That makes perfect sense @frosty ! Only one minor change to the above snippet, I think the condition for a blog that **can** upload video assets should be (if we put it in `Blog`):

```
func canUploadAsset(_ asset: WPMediaAsset) -> Bool {
    hasPaidPlan || !asset.exceedsFreeSitesAllowance()
}
```

does it sound right?

Reply:
Yep, that makes sense to me!